src/io/flutter/actions/OpenInAppCodeAction.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2021 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.actions;
+
+import com.intellij.execution.ExecutionException;
+import com.intellij.execution.configurations.GeneralCommandLine;
+import com.intellij.execution.process.ColoredProcessHandler;
+import com.intellij.execution.process.ProcessAdapter;
+import com.intellij.execution.process.ProcessEvent;
+import com.intellij.execution.process.ProcessOutputTypes;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.Presentation;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.project.ProjectUtil;
+import com.intellij.openapi.util.Key;
+import com.intellij.openapi.util.SystemInfo;
+import com.intellij.openapi.vfs.VirtualFile;
+import io.flutter.FlutterMessages;
+import io.flutter.sdk.FlutterSdk;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import static io.flutter.actions.OpenInXcodeAction.findProjectFile;
+
+public class OpenInAppCodeAction extends AnAction {
+
+  private static boolean IS_INITIALIZED = false;
+  private static boolean IS_APPCODE_INSTALLED = false;
+
+  static {
+    initialize();
+  }
+
+  private static void initialize() {
+    try {
+      // AppCode is installed if this shell command produces any output:
+      // mdfind "kMDItemContentType == 'com.apple.application-bundle'" | grep AppCode.app
+      final GeneralCommandLine cmd = new GeneralCommandLine().withExePath("/bin/bash")
```

Comment:
Does this run even if system is not mac?

Reply:
Good catch. There was supposed to be a check for that; I'll add it.